packages/twenty-server/src/engine/core-modules/ask-ai/ask-ai.service.ts

```diff
@@ -0,0 +1,181 @@
+import { LLMPromptTemplateService } from 'src/engine/integrations/llm-prompt-template/llm-prompt-template.service';
+import { LLMChatModelService } from 'src/engine/integrations/llm-chat-model/llm-chat-model.service';
+import { LLMTracingService } from 'src/engine/integrations/llm-tracing/llm-tracing.service';
+import { ObjectMetadataService } from 'src/engine/metadata-modules/object-metadata/object-metadata.service';
+import { computeTableName } from 'src/engine/utils/compute-table-name.util';
+
+@Injectable()
+export class AskAIService {
+  constructor(
+    private readonly workspaceDataSourceService: WorkspaceDataSourceService,
+    private readonly workspaceQueryRunnerService: WorkspaceQueryRunnerService,
+    private readonly llmChatModelService: LLMChatModelService,
+    private readonly llmPromptTemplateService: LLMPromptTemplateService,
+    private readonly llmTracingService: LLMTracingService,
+    private readonly objectMetadataService: ObjectMetadataService,
+  ) {}
+
+  private async getRecordMetadataById(
+    workspaceId: string,
+    workspaceDataSource: DataSource,
+    sqlQueryResult: Record<string, any>[],
+  ) {
+    const uuids = sqlQueryResult
+      .flatMap((row) => Object.values(row))
+      .filter((value) => isUUID(value, 4));
+
+    const objectMetadataEntities =
+      await this.objectMetadataService.findManyWithinWorkspace(workspaceId, {
+        where: { isSystem: false, isActive: true },
+      });
+
+    const columnNamesByStandardObjectName = {
+      opportunity: ['id', 'name'],
+      company: ['id', 'name', 'domainName'],
+      person: ['id', 'nameFirstName', 'nameLastName'],
+    };
+
+    const customObjectColumnNames = ['id', 'name'];
+
+    let recordMetadataById: RecordMetadataById = {};
+
+    for (const { nameSingular, isCustom } of objectMetadataEntities) {
+      const tableName = computeTableName(nameSingular, isCustom);
+
+      const columnNames = isCustom
+        ? customObjectColumnNames
+        : columnNamesByStandardObjectName[nameSingular];
+
+      const columnNamesString =
+        columnNames.map((col) => `"${col}"`).join(', ') || '';
+
+      // Security? GraphQL?
+      const recordMetadataSqlQuery = `SELECT ${columnNamesString} FROM "${tableName}" WHERE "id" = ANY($1);`;
+
+      const result: {
+        id: string;
+        name: string;
+        domainName?: string;
+        nameFirstName?: string;
+        nameLastName?: string;
+      }[] = await this.workspaceQueryRunnerService.executeSQL(
+        workspaceDataSource,
+        workspaceId,
+        recordMetadataSqlQuery,
+        [uuids],
+      );
+
+      recordMetadataById = result.reduce((recordMetadataById, row) => {
+        return {
+          ...recordMetadataById,
+          [row.id]: {
+            objectNameSingular: nameSingular,
+            ...omit(row, 'id'),
+          },
+        };
+      }, recordMetadataById);
+    }
+
+    return recordMetadataById;
+  }
+
+  async query(
```

Comment:
One challenge is pagination / how to handle a large number of results. It feels like there's a divergence between UI and CSV/download

Reply:
Could be for exemple, show count and limit to 30 results in view (because we enrich the data which is costly and poorly optimized). When downloading CSV view all results because we don't enrich?